### ISSUE TITLE
Q: txt file confusion

### ISSUE THREAD

**Issue by mergen3107** [koreader/koreader#10243]
I stumbled upon a book that I was able to find only in txt format. This is my first txt file opened in KOReader so I got so many questions:

1) Why does Style defaults to html/epub, if there is a Style for txt? Is it because it is located in Obsolete directory?

2) Default “publisher” font is monotype, everything is left-aligned, no hyphenations are applied. Is this intended of my txt file is broken?

3) There is no TOC, but I guess this because there no “headings”. Is there a possibility to try to get TOC by detecting patterns of `***`? Or something like that, this is a common “chapter divider”.

4) In the end I was able to get into readable state using Style Tweaks (page margins, fonts, sizes) to be able to use my beautiful profiles :D Is there a way to assign these style tweaks to the document defaults? For example, if I put all txt files into one folder. Or, in other way, a profile to apply style tweaks?

Thanks!

**Comment by poire-z** [comment]
All the .txt support there is is from the original crengine from 10+ years ago.
On the KOReader side, we haven't really been interested with the format - so no improvement, and not much understanding.
The existing code is supposed to handle various kind of txt formats, like full lines or 80 cols lines with arbitrary wraps, separator... and some russian specific formats like lib.ur.
All I had tried to do about it all these years is just to not break what's there :)

1) there was a txt.css, but it was full of crap: all the css were clone of each others, with then minor modifications that made them diverge for not good reason. We move them all in obsolete/.
After all, txt files end up converted to HTML with only 2 tags: `<pre>` and `<empty-line>`, so not much to style :) we put styles for these 2 tags in epub.css/html5.css , so they can be used with txt.

2) see issues #5042 #7293 #8216

3) Nope, unless it's taken care by the old crengine txt various kind of formats code. If your format is not detected as particular, not much else to do.
Don't ask me how to tweak your txt to have chapter detection working, I'd have to look at the txt code, which I don't really like and I'll hold a grudge against you :)

4) Probably, we have a plugin for that: #5428.
About profiles, you know their power better than us :)

TLDR: forget about txt :)

**Comment by mergen3107** [comment]
Thanks for explaining with patience :)
 And please forgive me for not looking it up first, I presumed txt to be much more obscure.

For this particular book it is OK, I restored my font face, size, and line spacing by disabling “publisher” style tweaks :D so we can probably close that one open issue you referenced in (2). 

I’ll better convert larger books to something sensible via calibre :D

**Comment by mergen3107** [comment]
!! Wow! I haven’t realized that there is this “Show in action menu” button on long-press of any style tweak! 

I was able to combine all txt-specific ones into a profile which I enable and disable via my profile gesture. Great!

The only thing I found is that when I disabled “Show in the action menu” for one of the style tweaks, it didn’t update the number of actions in the profile. So it still said “Edit actions (5)” instead of (4). 
Otherwise, profile still worked great :D
